finality-aleph/src/party/manager/mod.rs

```diff
@@ -404,20 +405,20 @@ where
     async fn early_start_validator_session(
         &self,
         session: SessionId,
+        node_id: NodeIndex,
```

Comment:
im not sure whether this should be provided externally. Previously we were sure whether node_id is a correct one. Now this might introduce some subtle bugs with idxes

Reply:
That's true, but there is another method here that already uses this approach (`spawn_authority_task_for_session`). Furthermore, the only use of this method went out of it's way to avoid learning about `node_id` which was pretty weird.

The Extremely Correct way of doing this would be to remove the `node_idx` method from the trait and have the methods do this internally. But this would also require merging the early start method with the nonvalidator method to an extent and in general would be a bigger (although not enormous) refactor.

I would suggest changing it here for consistency/simplicity and adding a task to refactor this properly, wdyt?

Reply:
I'm okay with that :)

Reply:
Added as A0-1769.